### PR TITLE
Init Dotnet workflows

### PR DESCRIPTION
**Pull request by nicolasbriere1**
Init Dotnet workflows
- Build and test workflow
- Publish workflow to public and private AWS nuget registries

### Build workflow : 
- Build dotnet project
- Unit test and coverage
- Pack in nuget 
- Upload to Github Packages in Nuget registry

### Publish workflow : 
- Pull from Github Packages
- Push to AWS and Nuget Public registry (Avoid breaking change to not  publish in public one for the moment)

### CODE REVIEW

**Review by Qlebrun — APPROVED** [review]

File: .github/workflows/dotnet-build.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+name: Dotnet - Build
+
+on:
+  workflow_call:
+    inputs:
+      dotnet_version:
+        required: false
+        type: string
+        default: "8.0.x"
+        description: "The dotnet version to use"
```
Comment (sGeeK44) [inline .github/workflows/dotnet-build.yml:10]: On peut lui donner plusieurs version ? (comme dans les autres projets)

File: .github/workflows/dotnet-build.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      project_name:
```
Comment (sGeeK44) [inline .github/workflows/dotnet-build.yml:11]: `project_name` => `solution_name` autant rester au plus près ça évite la confusion

File: .github/workflows/dotnet-build.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ inputs.dotnet_version }}
+
+      - name: Restore dependencies
+        run: dotnet restore ${{ inputs.project_name }}
```
Comment (sGeeK44) [inline .github/workflows/dotnet-build.yml:50]: En fait, cette étape ne sert à rien, elle est automatquement faite via le dotnet build si necessaire.

File: .github/workflows/dotnet-build.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+      - name: Add Coverage PR Comment
+        uses: marocchino/sticky-pull-request-comment@v2
+        if: github.event_name == 'pull_request' && inputs.code_coverage_enabled
+        with:
+          recreate: true
+          path: code-coverage-results.md
+
+      - name: Create Package PR
+        if: github.event_name == 'pull_request'
```
Comment (sGeeK44) [inline .github/workflows/dotnet-build.yml:109]: To move as input.

**Review by sGeeK44 — APPROVED** [review]
Beau boulot !

**Pull request merged**